src/components/menu/header-messages.jsx

```diff
@@ -53,7 +79,11 @@ export class HeaderMessagesUnconnected extends React.Component {
       case 'NEED_TO_LOGIN':
         content = (
           <span>
-            To save to this server, you need to <a onClick={this.props.login}>login</a>.
+            To save to this server, you need to
+            <a
+              onClick={this.showLoginModal.bind(this)}
```

Comment:
I don't think we want to bind this here-- you want to do that in the constructor, then just call it here via `onClick={this.showLoginModal()}`. Lots of examples of this elsewhere in iodide.